substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -205,10 +223,69 @@ private void abortIfSealed() {
         UserError.guarantee(!sealed, "Too late to add classes for serialization. Registration must happen in a Feature before the analysis has finished.");
     }
 
+    @Override
+    public void registerAllAssociatedClasses(ConfigurationCondition condition, Class<?> clazz) {
+        String targetClassName = clazz.getName();
+        // If the serialization target is primitive, it needs to get boxed, because the target is
+        // always an Object.
+        if (clazz.isPrimitive()) {
+            Class<?> boxedType = JavaKind.fromJavaClass(clazz).toBoxedJavaClass();
+            registerAllAssociatedClasses(condition, boxedType);
+            return;
+        } else if (!Serializable.class.isAssignableFrom(clazz)) {
+            println("Warning: The given class " + targetClassName + " is not serializable, it wouldn't get registered.\n");
+            return;
+        } else if (access.findSubclasses(clazz).size() > 1) {
+            println("Warning: The given class " + targetClassName +
+                            " has subclasses. We can't figure out which one should be registered. Skip.\n");
+            return;
+        }
+        try {
+            clazz.getDeclaredMethod("writeObject", ObjectOutputStream.class);
+            println("Warning: The given class " + targetClassName +
+                            " has customized serialization method writeObject. " +
+                            "We can't figure out what should be registered. Skip.\n");
+            return;
+        } catch (NoSuchMethodException e) {
+            // Expected case. Do nothing
+        }
+        register(condition, clazz);
+
+        if (clazz.isArray()) {
+            registerAllAssociatedClasses(condition, clazz.getComponentType());
+            return;
+        }
+        ObjectStreamClass osc = ObjectStreamClass.lookup(clazz);
+        try {
+            Method getDataLayoutMethod = ReflectionUtil.lookupMethod(ObjectStreamClass.class, "getClassDataLayout");
```

Comment:
Please also ensure this reflective method gets prepared in the SerializationBuilder constructor as you already did for `java.io.ObjectStreamClass$ClassDataSlot.desc`.

Reply:
Fixed